src/main/java/science/icebreaker/mail/MailService.java

```diff
@@ -31,6 +34,8 @@
     private final JavaMailSender mailSender;
     private final MailConfigurationProperties mailProperties;
 
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
```

Comment:
Cached thread pools do not have an upper bound on the number of threads created and mail sending is i/o bound. we will probably need a solid captcha for registration or/and we can just opt for a bounded thread pool

Reply:
Good point, I've changed it to a FixedThreadPool.